File: Api.Swazy/Api.Swazy/Models/Entities/Business.cs
using Api.Swazy.Models.Base;
using Api.Swazy.Types;

namespace Api.Swazy.Models.Entities;

public class Business : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string PhoneNumber { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Subtitle { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Footer { get; set; } = string.Empty;
    public string Theme { get; set; } = string.Empty;
    public BusinessType BusinessType { get; set; }
    public string WebsiteUrl { get; set; } = string.Empty;
    
    public virtual List<UserBusinessAccess> UserAccesses { get; set; } = new();
    public virtual List<BusinessService> Services { get; set; } = new();
}

File: swazy-ui/src/app/models/business.ts
import {Employee} from './employee';
import {Service} from './service';

export interface Business {
  id: string;
  name: string;
  address: string;
  phoneNumber: string;
  email: string;
  title: string;
  subtitle: string;
  description: string;
  footer: string;
  theme: string;
  businessType: string;
  websiteUrl: string;
  employees?: Employee[];
  services?: BusinessService[];
  createdAt?: string;
}

export interface BusinessService {
  id: string;
  businessId: string;
  serviceId: string;
  serviceName: string;
  price: number;
  duration: number;
  createdAt: string;
}
